Updated Dependencies. Refractor SEC web scraper. Implemented date range support.

brandovlee opened the pull request:
### Summary ### 
The code was refactored for better readability by modularizing functions and updating dependencies to the latest versions. Additionally, navigation and selectors were enhanced, enabling the scraper to retrieve missing data from multiple days.


### TODO ###
- [x] Upload Docker image to ECR and set ECS task back to running every day
- [x] Turn Cloudwatch Alarm back on

s-egge commented on SEC/readSEC.js:
```diff
@@ -91,130 +85,342 @@ const axios = require("axios");
+      const PVTable = {
+        meterName,
+        meterID,
+        END_TIME,
+        END_TIME_SECONDS,
+        PVSystem,
+        totalDailyYield,
```
```suggestion
        time: END_TIME,
        time_seconds: END_TIME_SECONDS,
        PVSystem,
        totalYield: totalDailyYield,
```

Make sure this is formatted correctly for the backend!

s-egge commented on SEC/readSEC.js:
```diff
@@ -91,130 +85,342 @@ const axios = require("axios");
+ * Returns the last date that data was logged to the dashboard
+ */
+async function getLastLoggedDate() {
+  return "11/04/2024"; // TODO: implement a GET request to the API to get the last logged date
```
Same as the `Ennex` scraper, let's make this return yesterday's date for now so that it runs essentially the same as before (gets yesterday's date) and we can upload a version to AWS with the upload code commented out to make sure that the dates are still right with the timezone differences.

s-egge reviewed (changes requested):
It's working well! It just needs a few changes and it will be good to merge.

s-egge commented:
There is a small formatting issue with the date output, but otherwise it looks like it's working well both locally and on AWS now that we've changed the ECS timezone to ours. 

```
Actual Date: 09/03/
Date: 09/03/ | Energy: 127.35
 {
  meterName: 'SEC Solar',
  meterID: 123,
  time: '2025-03-09T23:59:59',
  time_seconds: 1741589999,
  PVSystem: 'SECsolar',
  totalYield: '127.35'
}
```

brandovlee commented:
> There is a small formatting issue with the date output, but otherwise it looks like it's working well both locally and on AWS now that we've changed the ECS timezone to ours.
> 
> ```
> Actual Date: 09/03/
> Date: 09/03/ | Energy: 127.35
>  {
>   meterName: 'SEC Solar',
>   meterID: 123,
>   time: '2025-03-09T23:59:59',
>   time_seconds: 1741589999,
>   PVSystem: 'SECsolar',
>   totalYield: '127.35'
> }
> ```

If this is referring to the console.log(date), it's sourced from the SEC dashboard. We can either choose to keep it as is or format it to make it more readable (but then it won't align with the way that it's displayed on the website)

<img width="648" alt="image" src="https://github.com/user-attachments/assets/b4727e1c-c270-46ed-a4fe-b20382626a31" />

s-egge commented:
> > There is a small formatting issue with the date output, but otherwise it looks like it's working well both locally and on AWS now that we've changed the ECS timezone to ours.
> > ```
> > Actual Date: 09/03/
> > Date: 09/03/ | Energy: 127.35
> >  {
> >   meterName: 'SEC Solar',
> >   meterID: 123,
> >   time: '2025-03-09T23:59:59',
> >   time_seconds: 1741589999,
> >   PVSystem: 'SECsolar',
> >   totalYield: '127.35'
> > }
> > ```
> 
> If this is referring to the console.log(date), it's sourced from the SEC dashboard. We can either choose to keep it as is or format it to make it more readable (but then it won't align with the way that it's displayed on the website)
> <img alt="image" width="648" src="https://private-user-images.githubusercontent.com/165731814/421159229-b4727e1c-c270-46ed-a4fe-b20382626a31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE3MTI3NDgsIm5iZiI6MTc0MTcxMjQ0OCwicGF0aCI6Ii8xNjU3MzE4MTQvNDIxMTU5MjI5LWI0NzI3ZTFjLWMyNzAtNDZlZC1hNGZlLWIyMDM4MjYyNmEzMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxMVQxNzAwNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NjIwOGFkYjYwZjZlZTQ5MjE3MmZhNmIzZjdjZWMyMjc1MjRlNjIwYmRhYmNiN2NlMzZiNjg5MTdhMmE5MzAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rDl-29Z2ZPQehXxCeMzS-p1Eg2h5ktWCZ9iuHEZvrFs">

I see, I think that since logging is for us to see what's going on, we should make it clearer. It looked like September 3rd to me when I first saw it, and also seemed like it was missing data with the trailing `/`. We can do something like swapping the numbers when logging so it would read like `03/09`, and then if there's a mismatch we can log the website version for debugging (something like `Error: Expected selector 09/03/ but found 10/04/` ). 

We should also fix the response logs to how we updated [Ennex logs](https://github.com/OSU-Sustainability-Office/automated-jobs/pull/84/files#diff-82e3724a6155fa26fbca942d5fc938c0837f4136d58f704dea23db8a1fe7d25bL375).

brandovlee commented:
> > > There is a small formatting issue with the date output, but otherwise it looks like it's working well both locally and on AWS now that we've changed the ECS timezone to ours.
> > > ```
> > > Actual Date: 09/03/
> > > Date: 09/03/ | Energy: 127.35
> > >  {
> > >   meterName: 'SEC Solar',
> > >   meterID: 123,
> > >   time: '2025-03-09T23:59:59',
> > >   time_seconds: 1741589999,
> > >   PVSystem: 'SECsolar',
> > >   totalYield: '127.35'
> > > }
> > > ```
> > 
> > 
> > If this is referring to the console.log(date), it's sourced from the SEC dashboard. We can either choose to keep it as is or format it to make it more readable (but then it won't align with the way that it's displayed on the website)
> > <img alt="image" width="648" src="https://private-user-images.githubusercontent.com/165731814/421159229-b4727e1c-c270-46ed-a4fe-b20382626a31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE3MTI3NDgsIm5iZiI6MTc0MTcxMjQ0OCwicGF0aCI6Ii8xNjU3MzE4MTQvNDIxMTU5MjI5LWI0NzI3ZTFjLWMyNzAtNDZlZC1hNGZlLWIyMDM4MjYyNmEzMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxMVQxNzAwNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NjIwOGFkYjYwZjZlZTQ5MjE3MmZhNmIzZjdjZWMyMjc1MjRlNjIwYmRhYmNiN2NlMzZiNjg5MTdhMmE5MzAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rDl-29Z2ZPQehXxCeMzS-p1Eg2h5ktWCZ9iuHEZvrFs">
> 
> I see, I think that since logging is for us to see what's going on, we should make it clearer. It looked like September 3rd to me when I first saw it, and also seemed like it was missing data with the trailing `/`. We can do something like swapping the numbers when logging so it would read like `03/09`, and then if there's a mismatch we can log the website version for debugging (something like `Error: Expected selector 09/03/ but found 10/04/` ).
> 
> We should also fix the response logs to how we updated [Ennex logs](https://github.com/OSU-Sustainability-Office/automated-jobs/pull/84/files#diff-82e3724a6155fa26fbca942d5fc938c0837f4136d58f704dea23db8a1fe7d25bL375).

I just updated the debug output date format so that it's easier to read. I'll leave screenshots of the updated output. I also modified the response output to make it easier to read the logs, exactly like the implementation with ENNEX. I've also uploaded the latest changes to AWS with the scheduler so this PR is ready to merge if everything looks good. 

<img width="258" alt="Screenshot 2025-03-11 at 1 52 49 PM" src="https://github.com/user-attachments/assets/994d8bb7-5dec-4d3c-9159-59ccac769be4" />
<img width="450" alt="Screenshot 2025-03-11 at 1 51 51 PM" src="https://github.com/user-attachments/assets/53fc1bbd-9ff3-4407-a932-e0de70f08e66" />

s-egge reviewed (approved):
Looks great!
